R/gitlab.R

```diff
@@ -71,7 +71,7 @@ gitlabArchiveUrl <- function(pkgRecord) {
   archiveUrl <- sprintf(fmt,
                         pkgRecord$remote_host,
                         pkgRecord$remote_username,
-                        pkgRecord$remote_repo,
+                        sub("/", "%2F", pkgRecord$remote_repo),
```

Comment:
Thanks, @galachad

Note to @toph-allen --

Per https://docs.gitlab.com/ee/api/repositories.html#get-file-archive, the `:id` needs to be URL-encoded. We may want to use this opportunity to escape other characters, as well.

```r
URLencode(pkgRecord$remote_repo, TRUE)
```

Additionally, we should update NEWS.

Reply:
Yeah, we should use [`URLencode`](https://stat.ethz.ch/R-manual/R-devel/library/utils/html/URLencode.html) here, I think.

I'm also going to check GitHub and Bitbucket to see if we should treat them similar.